### DISCUSSION TITLE
Pretty URL's cause videos not to work.

### DISCUSSION THREAD

**Discussion by taneharre** [GetPublii/Publii#1766] [Help]
Does anyone else have a problem with videos not working if Pretty URL's are enabled? I have been having a lot of problems with videos not loading or giving the error "No video with supported format and MIME type found" . They would always work in preview but not after uploading to my hosting. I am using the standard video block, 300 * 150 with webm files. The problem is cured if I change to non pretty URL's.

Additional info: Nginx, password protected site.

**Comment by dziudek** [comment]
@taneharre - could you describe how the URL looks on preview and how it is generated on the production page? We didn't have such issues reported before

**Comment by taneharre** [comment]
Version: 0.46.1 (build 16918)
Theme: Mercury v3.0.0.0
LMDE 6

Like this. Erasing is all mine though. :)
## Not Pretty URLs preview
![Not Pretty URLs preview](https://github.com/user-attachments/assets/54ae937a-cb16-4c15-9a33-2ab88cb04327)

## Not Pretty URLs production
![Not Pretty URLs production](https://github.com/user-attachments/assets/0dd8330c-dbae-440e-91d9-5a8fed385c36)

## Pretty URLs preview
![Pretty URLs preview](https://github.com/user-attachments/assets/35de046b-892c-4f0c-b0d8-ac58ae459f82)


## Pretty URLs production
![Pretty URLs production](https://github.com/user-attachments/assets/2f990ad8-fa33-400f-9b97-cc57d423b990)

**Reply by dziudek** [reply]
@taneharre - strange - it seems that first frame is loaded correctly, so browser is able to access the video. 

Could you put here how the code of the video element looks on the preview and on the production website - please replace domain name with `example.com` - I will try to diagnose the reason of the issue according to this

**Reply by taneharre** [reply]
There is one more thing about the site that might make all the difference. It is password protected and not with .htpasswd as it is on Nginx.

# Preview

`<h2>Piano Lesson</h2><figure class="post__video"><video loading="lazy" width="300" height="150" controls="controls" data-mce-fragment="1" poster="file:///home/example/example/Publii/Public/media/posts/19/2020-04-10-12.31.10_2.jpg"><source src="media/files/2020-04-10-12.31.10_2.webm" type="video/webm"></video></figure>`

# Production

`<h2>Piano Lesson</h2><figure class="post__video"><video loading="lazy" width="300" height="150" controls="controls" data-mce-fragment="1" poster="https://example.com/media/posts/19/2020-04-10-12.31.10_2.jpg"><source src="media/files/2020-04-10-12.31.10_2.webm" type="video/webm"></video></figure>`

**Reply by dziudek** [reply]
@taneharre - there is an answer - source element contains relative path to the file - what probably causes the issue. Please try to put there the full path. Probably it won't work on preview, but should work on production
